nilearn/region/AutoExtractor.py

```diff
@@ -0,0 +1,132 @@
+"""
+Auto Extractor
+"""
+
+import numpy as np
+
+from scipy.ndimage.measurements import label
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.externals.joblib import Memory
+
+from nilearn._utils import check_niimg
+
+
+def compute_low_threshold(maps, ratio):
+    """
+    Parameters
+    ----------
+    maps: numpy array like of
+        a 3D/4D Niimg-like object which contains data of the
+        input maps
+
+    ratio: a float value
+        this value is used to apply on the maps
+
+    Returns
+    -------
+    a lower threshold value computed from the maps
+
+    """
+    if not isinstance(maps, np.ndarray):
+        raise ValueError("Given input maps should be an array like")
+
+    maps_ravel = np.abs(maps).ravel()
+    argsort = np.argsort(maps_ravel)
+    n_voxels = int(ratio * maps[0].size)
+    low_threshold = maps_ravel[argsort[- n_voxels]]
+    return low_threshold
+
+
+def compute_regions(maps):
+    """
+    Parameters
+    ----------
+    maps: 3D/4D Nifti like object
+
+    Returns
+    -------
+    the regions which are strongly connected components
+    from the maps
+
+    """
+    label_maps = []
+    for map_ in maps:
+        if np.all(map_ == 0):
```

Comment:
``np.any`` conversion of the code might be faster (although probably not relevant here)